src/components/views/rooms/RoomTileIcon.tsx

```diff
@@ -68,17 +60,27 @@ export default class RoomTileIcon extends React.Component<IProps, IState> {
         return joinRule === 'public';
     }
 
-    public componentWillUnmount() {
-        this.isUnmounted = true;
-        if (this.isWatchingTimeline) this.props.room.off('Room.timeline', this.onRoomTimeline);
-        this.unsubscribePresence();
+    private get dmUser(): User {
+        return this._dmUser;
     }
 
-    private unsubscribePresence() {
-        if (this.dmUser) {
-            this.dmUser.off('User.currentlyActive', this.onPresenceUpdate);
-            this.dmUser.off('User.presence', this.onPresenceUpdate);
+    private set dmUser(val: User) {
```

Comment:
this would be so much cleaner as a hook :'(

Reply:
I've yet to be convinced on hooks tbh

Reply:
Fair, just that this would be much shorter, something like

`useEventEmitter(this.props.room, "Room.timeline", this.onRoomTimeline)`

and it'd handle the on and off for you magically